doc/conf.py

```diff
@@ -68,6 +68,55 @@
 nbsphinx_kernel_name='python'
 nbsphinx_timeout = 60
 
+# This is processed by Jinja2 and inserted before each notebook
+nbsphinx_prolog = r"""
+{% set docname = 'doc/' + env.doc2path(env.docname, base=None) %}
+{% set git_ref = 'master' if '.' not in env.config.current_version else 'v' + env.config.release %}
+.. raw:: html
+
+    <div class="admonition note">
+      <p>This page was generated from
+        <a class="reference external" href="https://github.com/pygae/galgebra/blob/{{ git_ref|e }}/{{ docname|e }}">{{ docname|e }}</a>.
+        <!--
+            This does not work yet due to nbsphinx-link
+            Interactive online version:
+            <a href="https://mybinder.org/v2/gh/pygae/galgebra/{{ git_ref|e }}?filepath={{ docname|e }}"><img alt="Binder badge" src="https://mybinder.org/badge_logo.svg" style="vertical-align:text-bottom"></a>.
+        -->
+      </p>
+      <script>
+        if (document.location.host) {
+          var p = document.currentScript.previousSibling.previousSibling;
+          var a = document.createElement('a');
+          a.innerHTML = 'View in <em>nbviewer</em>';
+          a.href = `https://nbviewer.jupyter.org/url${
```

Comment:
I definitely prefer a github based nbviewer url. Any particular reason/difficulty made you choose the URL based approach?

Reply:
Mainly because it meant I could copy-paste the approach from clifford.

Reply:
It renders a lot faster too.